internal/broadcaster/multi_utxo_creator.go

```diff
@@ -0,0 +1,59 @@
+package broadcaster
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+	"time"
+)
+
+type MultiKeyUTXOCreator struct {
+	creators    []*UTXOCreator
+	logger      *slog.Logger
+	cancelAll   context.CancelFunc
+	ctx         context.Context
+	wg          sync.WaitGroup
```

Comment:
With waiting logic moved inside UTXO creator these field won't be needed anymore